### PR TITLE
fix: [Table] field named `data` will produce bugged output

### LINKED ISSUE #8051: Bug: HTML Table Class fails with table column named 'data'.
### PHP Version

8.2

### CodeIgniter4 Version

4.4.1

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

apache

### Database

mariadb  Ver 15.1 Distrib 10.11.4-MariaDB, for debian-linux-gnu (x86_64) using  EditLine wrapper

### What happened?

HTML Table Class produces bugged output as follows:

<table border="1" cellpadding="2" cellspacing="1"> <thead> <tr> <th valor_desconto="Valor do Desconto" tipo_desconto="Tipo de Desconto" codigo="Codigo Orçamento">Data do Orçamento</th></tr> </thead> <tbody> <tr> <td deleted_at="" updated_at="2023-10-16 21:12:29" created_at="2023-10-16 21:12:29" valor_desconto="" tipo_desconto="NENHUM" codigo="codigo1" id_cliente="1" id="1">2023-10-16 21:12:29</td></tr> <tr> <td deleted_at="" updated_at="2023-10-16 21:12:29" created_at="2023-10-16 21:12:29" valor_desconto="10.00" tipo_desconto="REAL" codigo="codigo2" id_cliente="2" id="2">2023-10-16 21:12:29</td></tr> <tr> <td deleted_at="" updated_at="2023-10-16 21:12:29" created_at="2023-10-16 21:12:29" valor_desconto="10.00" tipo_desconto="PERCENTUAL" codigo="codigo3" id_cliente="3" id="3">2023-10-16 21:12:29</td></tr> </tbody> </table>

### Steps to Reproduce

Serve Model->asArray()->findAll() to generate() method of HTML Table class when source table contains column named "data".

### Expected Output

<table border="1" cellpadding="2" cellspacing="1"> <thead> <tr> <th>Codigo Orçamento</th><th>Data do Orçamento</th><th>Tipo de Desconto</th><th>Valor do Desconto</th></tr> </thead> <tbody> <tr> <td>codigo1</td><td>2023-10-16 21:50:08</td><td>NENHUM</td><td></td></tr> <tr> <td>codigo2</td><td>2023-10-16 21:50:08</td><td>REAL</td><td>10.00</td></tr> <tr> <td>codigo3</td><td>2023-10-16 21:50:08</td><td>PERCENTUAL</td><td>10.00</td></tr> </tbody> </table>

### Anything else?

ci4-html_table_class-bug_report

### PR DESCRIPTION
**Pull request by ping-yee**
**Description**
Fixes #8051 

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
@ping-yee Can you add test code?

**Comment by ping-yee** [comment]
@kenjis Done.

File: system/View/Table.php [hunk 1]
```diff
@@ -265,7 +265,7 @@ protected function _prepArgs(array $args)
         // If there is no $args[0], skip this and treat as an associative array
         // This can happen if there is only a single key, for example this is passed to table->generate
         // array(array('foo'=>'bar'))
-        if (isset($args[0]) && count($args) === 1 && is_array($args[0]) && ! isset($args[0]['data'])) {
```
Comment (kenjis) [inline system/View/Table.php:268]: Why was the `data` excluded?

Comment (ping-yee) [reply system/View/Table.php:268]: It will cause the `setHeading` function to be executed incorrectly when the heading field contain named `data`.

Comment (kenjis) [reply system/View/Table.php:268]: The `&& ! isset($args[0]['data'])` came from this commit: https://github.com/bcit-ci/CodeIgniter/commit/f0b69a85
Are you absolutely sure that just removing it does not cause another issue?
Frankly, I don't get the code yet.

Comment (ping-yee) [reply system/View/Table.php:268]: I am not sure but the all of the test cases get passed.
And I also don't get where will use this condition and the data processing of the array field contains `data` so far.

Or we can add one more process to check that whether the heading array has been handled correctly when the table is being generated.

Comment (kenjis) [reply system/View/Table.php:268]: > the all of the test cases get passed.

Yes. But we cannot see the coverage in Coveralls.

> coverage/coveralls — Target branch is ahead of PR base. Rebase the PR to fix. 

Can you rebase the PR branch?

**Comment by kenjis** [comment]
If anyone has a good understanding of the Table's code, please review this PR.

File: system/View/Table.php (hunk 1 above)
Comment (ping-yee) [reply system/View/Table.php:268]: > Can you rebase the PR branch?

Done.

Comment (dgvirtual) [reply system/View/Table.php:268]: Based on my usage of the table class (quite extensive) I have not noticed any issues thus far with the patch applied.

**Comment by kenjis** [comment]
This bug has been introduced in:
https://github.com/codeigniter4/CodeIgniter4/commit/c7b02d89#diff-2ca8e500cfe5125fe8fb57a83cdf78c178594f553e4639012bef5a23634c7e97L439-R482

File: tests/system/View/TableTest.php [hunk 2]
```diff
@@ -816,6 +816,99 @@ public static function orderedColumnUsecases(): iterable
...
+
+            'row_alt_start'  => '<tr>',
+            'row_alt_end'    => '</tr>',
+            'cell_alt_start' => '<td>',
+            'cell_alt_end'   => '</td>',
+
+            'table_close' => '</table>',
+        ];
+
+        $table = new Table($tableTemplate);
```
Comment (kenjis) [inline tests/system/View/TableTest.php:860]:
```suggestion
        $table = new Table();
```

**Review by kenjis — APPROVED** [review]

File: system/View/Table.php (hunk 1 above)
Comment (kenjis) [reply system/View/Table.php:268]: Thank you all!

**Pull request merged**